pyne/alara.py

```diff
@@ -287,31 +289,51 @@ def record_to_geom(mesh, cell_fracs, cell_mats, geom_file, matlib_file,
     mixture = '' # mixture blocks
 
     unique_mixtures = []
-    for i, mat, ve in mesh:
-        volume += '    {0: 1.6E}    zone_{1}\n'.format(mesh.elem_volume(ve), i)
-
-        ve_mixture = {}
-        for row in cell_fracs[cell_fracs['idx'] == i]:
-            cell_mat = cell_mats[row['cell']]
-            name = cell_mat.metadata['name']
-            if _is_void(name):
-                name = 'mat_void'
-            if name not in ve_mixture.keys():
-                ve_mixture[name] = np.round(row['vol_frac'], sig_figs)
-            else:
-                ve_mixture[name] += np.round(row['vol_frac'], sig_figs)
-
-        if ve_mixture not in unique_mixtures:
-            unique_mixtures.append(ve_mixture)
-            mixture += 'mixture mix_{0}\n'.format(
-                                           unique_mixtures.index(ve_mixture))
-            for key, value in ve_mixture.items():
-                mixture += '    material {0} 1 {1}\n'.format(key, value)
-
-            mixture += 'end\n\n'
+    if not sub_voxel:
```

Comment:
I think this conditional can be at least one layer deeper in the loops, esp. once you have reordered the loops for the subvoxel case.

It may be appropriate to use the conditional in many separate places rather than have two very similar blocks.

Reply:
Although these two blocks are both used to materials information to ALARA input file, but actually they have a big different. 

For the normal R2S, we need to name the `mixture` composed of different materials, and assign a id to this `mixture`, however, we don't need to do that in sub-voxel R2S. Because the material of each sub-voxel is the same as corresponding cell. So I write `material.name` to ALARA input file to simplify this process. 

These two blocks look similar but do writing different content. It would make the code more complex if we make the conditional be deeper in the loops, because we need to make conditional judgement for almost every line of the code.